api/v1alpha1/backendsecurity_policy.go

```diff
@@ -175,10 +174,17 @@ type BackendSecurityPolicyGCPCredentials struct {
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:MinLength=1
 	Region string `json:"region"`
-	// WorkLoadIdentityFederationConfig is the configuration for the GCP Workload Identity Federation.
+
+	// ServiceAccountImpersonation is the service account impersonation configuration.
+	// This is used to impersonate a service account when getting access token.
+	//
+	// +optional
+	ServiceAccountImpersonation *GCPServiceAccountImpersonationConfig `json:"serviceAccountImpersonation,omitempty"`
```

Comment:
@sukumargaonkar i read a little more on this, we might need to move impersonation back into the federation section as this method always requires an original identity to impersonate. The other method is the static service account token file which you can download from gcp console to authenticate.